app/controllers/users_controller.rb

```diff
@@ -209,6 +209,49 @@ def check_emails
     render json: failed_json, status: 403
   end
 
+  def update_primary_email
+    params.require(:email)
+
+    user = fetch_user_from_params
+    guardian.ensure_can_edit!(user)
+
+    if user.primary_email.email == params[:email]
+      return render json: success_json
+    end
+
+    new_primary = user.user_emails.find_by(email: params[:email])
+    if new_primary.blank?
+      return render json: failed_json, status: 428
```

Comment:
Looks like we never used that HTTP status before. How does the UI handle that?

Reply:
It should only happen if the user deletes the same email address. I will show a bootbox if this happens.